script.js

```diff
@@ -0,0 +1,987 @@
+            isShift: ['▼','▼',],
+            caps: ['▼','▼',],
+            shiftCaps: ['▼','▼',],
+        },
+    },
+    {
+        code: 'ArrowRight',
+        rus_eng: {
+            noShift: ['►','►',],
+            isShift: ['►','►',],
+            caps: ['►','►',],
+            shiftCaps: ['►','►',],
+        },
+    },
+    {
+        code: 'ControlRight',
+        rus_eng: {
+            noShift: ['Ctrl','Ctrl',],
+            isShift: ['Ctrl','Ctrl',],
+            caps: ['Ctrl','Ctrl',],
+            shiftCaps: ['Ctrl','Ctrl',],
+        },
+    },
+];
+
+//const allKeyArray = buttonArrayRow1.concat(buttonArrayRow2, buttonArrayRow3, buttonArrayRow4, buttonArrayRow5);
+
+class V_keyboard {
+    constructor () {
+        this.capsLock = false;
+        this.shift = false;
+        this.alt = false;
+        this.lang = localStorage.getItem('lang') || 'eng';
+        this.textArea = document.createElement('textarea');
+    }
+
+    createButtonRow(keyboard, buttonArrayRow) {
+        const row1 = document.createElement('div');
+        row1.classList.add('row');
+        keyboard.append(row1);
+
+        for (let button  of buttonArrayRow) {
+            let key = document.createElement('div');
+            key.classList.add('key',button.code);
+            key.dataset.button = button.code;
+
+            row1.append(key);
+
+            let span_rus = document.createElement('span');
+            let span_eng = document.createElement('span');
+
+            if (this.lang === 'rus') {
+                span_rus.classList.add('rus');
+                span_eng.classList.add('eng', 'hidden');
+            } else {
+                span_rus.classList.add('rus', 'hidden');
+                span_eng.classList.add('eng');
+            }
+
+            key.append(span_eng);
+            key.append(span_rus);
+
+            const span_rus_var = `<span class="noShift hidden">${button.rus_eng.noShift[0]}</span>
+                                  <span class="isShift hidden">${button.rus_eng.isShift[0]}</span>
+                                  <span class="caps hidden">${button.rus_eng.caps[0]}</span>
+                                  <span class="shiftCaps hidden">${button.rus_eng.shiftCaps[0]}</span>`;
+
+            const span_eng_var = `<span class="noShift">${button.rus_eng.noShift[1]}</span>
+                                  <span class="isShift hidden">${button.rus_eng.isShift[1]}</span>
+                                  <span class="caps hidden">${button.rus_eng.caps[1]}</span>
+                                  <span class="shiftCaps hidden">${button.rus_eng.shiftCaps[1]}</span>`;
+
+            span_rus.innerHTML = span_rus_var;
+            span_eng.innerHTML = span_eng_var;
+
+            key.addEventListener('click', this.clickButton.bind(this));
+        }
+
+    }
+
+    init() {
+        const wrapper = document.createElement('div');
+        wrapper.classList.add('wrapper');
+
+        document.body.append(wrapper);

```

Comment:
The recommended practice is to build all of your elements and THEN append them in one step (if possible). This is the best from a performance perspective.
When appending a series of elements to the DOM each triggers the browser to reflow the page. Adding one element to the page that has already been prepared while not on the DOM then appending that to the DOM only causes one reflow of the page. https://www-archive.mozilla.org/newlayout/doc/reflow.html

So this row should be moved to the end of the method

Reply:
done